### ISSUE TITLE
[Health] Could not cast value of type 'HKWorkoutType' to 'HKQuantityType'

### ISSUE THREAD

**Issue by obrunsmann** [cph-cachet/flutter-plugins#463]
### Device / Emulator and OS
Please complete the following information for each phone and/or emulator you're experiencing this bug on:
 - Device: iPhone XS
 - OS: iOS 15.02.

### Describe the bug
Crash when saving health data.

### To Reproduce
```dart
healthFactory.writeHealthData(
  10.0,
  HealthDataType.WORKOUT,
  DateTime.now().subtract(duration),
  DateTime.now(),
);
```

### Expected behavior
Stores the workout

### Actual behavior
```
Successfully called writeData with value of 10.0 and type of WORKOUT
2
Could not cast value of type 'HKWorkoutType' (0x1da0bb930) to 'HKQuantityType' (0x1da0bb890).
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001b71c69c4 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1b71c69c4 <+8>:  b.lo   0x1b71c69e4               ; <+40>
    0x1b71c69c8 <+12>: pacibsp
    0x1b71c69cc <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1b71c69d0 <+20>: mov    x29, sp
Target 0: (Runner) stopped.
Lost connection to device.
```

### Flutter doctor 
```
[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-x64, locale de-DE)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] VS Code (version 1.62.3)
[✓] Connected device (3 available)
```

**Comment by drodenberg** [comment]
Any news on this issue? I'm seeing the exact same thing when I try and writeHealthData to the Mindfulness minutes. The `health.requestAuthorization` function is returning true with regards to the request being granted.

**Comment by drodenberg** [comment]
I found this... https://github.com/cph-cachet/flutter-plugins/blob/765796cdecef60e1ec08af7dc6308f5e727697f6/packages/health/ios/Classes/SwiftHealthPlugin.swift#L375

According to that line your app needs to be on ios 11 or later. I had to change the lowest version to be 11 and it started working for me!

**Comment by obrunsmann** [comment]
We are already working on a PR here. The implementation is almost finished so far, after we are back from Christmas vacation we will test some more.

**Comment by drodenberg** [comment]
```Swift
let sample: HKObject
      
      if (unitLookUp(key: type) == HKUnit.init(from: "")) {
          sample = HKCategorySample(type: dataTypeLookUp(key: type) as! HKCategoryType, value: Int(value), start: dateFrom, end: dateTo)
        } else {
          let quantity = HKQuantity(unit: unitLookUp(key: type), doubleValue: value)
          
          sample = HKQuantitySample(type: dataTypeLookUp(key: type) as! HKQuantityType, quantity: quantity, start: dateFrom, end: dateTo)
        }
```

**Comment by drodenberg** [comment]
line 192 SwiftHealthPlugin
This change in the SwiftHealthPlugin fixed everything for me.

**Comment by drodenberg** [comment]
This PR should fix this issue. I've tested it on my side. https://github.com/cph-cachet/flutter-plugins/pull/485

**Comment by bardram** [comment]
Fixed (addressed) as part of PR #485 and released as version 3.4.1. 

Thanks to @drodenberg  for the PR.
